azurerm/resource_arm_storage_account.go

```diff
@@ -963,3 +972,124 @@ func flattenAzureRmStorageAccountIdentity(identity *storage.Identity) []interfac
 
 	return []interface{}{result}
 }
+
+func flattenAzureRmStorageAccountPrimaryEndpoints(d *schema.ResourceData, primary *storage.Endpoints, acctName *string, acctKey *string) error {
+	if primary == nil {
+		return fmt.Errorf("primary endpoints should not be empty")
+	}
+
+	if acctName == nil {
+		return fmt.Errorf("account name should not be nil")
+	}
+
+	if acctKey == nil {
+		return fmt.Errorf("account key should not be nil")
+	}
+
+	var blobEndpoint, blobHost, blobConnectionStr string
+	if v := primary.Blob; v != nil {
+		blobEndpoint = *v
+
+		if u, err := url.Parse(*v); err != nil {
+			blobHost = u.Host
+		} else {
+			return fmt.Errorf("invalid blob endpoint for parsing: %q", *v)
+		}
+
+		blobConnectionStr = fmt.Sprintf("DefaultEndpointsProtocol=https;BlobEndpoint=%s;AccountName=%s;AccountKey=%s", blobEndpoint, *acctName, *acctKey)
+	}
+	d.Set("primary_blob_endpoint", blobEndpoint)
+	d.Set("primary_blob_host", blobHost)
+	d.Set("primary_blob_connection_string", blobConnectionStr)
+
+	var queueEndpoint, queueHost string
+	if v := primary.Queue; v != nil {
+		queueEndpoint = *v
+
+		if u, err := url.Parse(*v); err != nil {
+			queueHost = u.Host
```

Comment:
Should this be:
```suggestion
			return err;
```
instead of setting the host when `err` is not `nil`?

Reply:
@JunyiYi Thanks for pointing it out, I've updated the PR.